src/models/Activity.js

```diff
@@ -45,7 +45,7 @@ ActivitySchema.virtual('quizCount').get(setQuizCount);
 
 // eslint-disable-next-line consistent-return
 function setAreCardsValid() {
-  if (this.cards && this.cards.length === 0) return true;
+  if (this.cards && this.cards.length === 0) return false;
```

Comment:
il faudrait changer la condition en ` if (!this.cards || this.cards.length === 0) return false;` car si l'objet `cards` n'est pas defini ce n'est pas valide